js/event.ts

```diff
@@ -0,0 +1,282 @@
+  private _path: domTypes.EventPath[] = [];
+
+  constructor(type: string, eventInitDict?: domTypes.EventInit) {
+    eventAttributes.set(this, {
+      type,
+      bubbles: eventInitDict && eventInitDict.bubbles || false,
+      cancelable: eventInitDict && eventInitDict.cancelable || false,
+      composed: eventInitDict && eventInitDict.composed || false,
+      currentTarget: null,
+      eventPhase: domTypes.EventPhase.NONE,
+      isTrusted: false,
+      target: null,
+      timeStamp: Date.now(),
+    });
+  }
+
+  get bubbles(): boolean {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).bubbles || false;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get cancelBubble(): boolean {
+    return this._stopPropagationFlag;
+  }
+
+  get cancelBubbleImmediately(): boolean {
+    return this._stopImmediatePropagationFlag;
+  }
+
+  get cancelable(): boolean {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).cancelable || false;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get composed(): boolean {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).composed || false;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get currentTarget(): domTypes.EventTarget {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).currentTarget || null;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get defaultPrevented(): boolean {
+    return this._canceledFlag;
+  }
+
+  get eventPhase(): number {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).eventPhase || domTypes.EventPhase.NONE;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get isTrusted(): boolean {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).isTrusted || false;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get target(): domTypes.EventTarget {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).target || null;
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get timeStamp(): Date {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).timeStamp || Date.now();
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  get type(): string {
+    if (eventAttributes.has(this)) {
+      return eventAttributes.get(this).type || "";
+    }
+
+    throw new TypeError("Illegal invocation");
+  }
+
+  /** Returns the event’s path (objects on which listeners will be
+   * invoked). This does not include nodes in shadow trees if the
+   * shadow root was created with its ShadowRoot.mode closed.
+   *
+   *      event.composedPath();
+   */
+  composedPath(): domTypes.EventPath[] {
+    if (this._path.length === 0) {
+      return [];
+    }
+
+    const composedPath: domTypes.EventPath[] = [{
```

Comment:
With the cast below, `as domTypes.EventPath`, this type assertion is redundant.  One other the other should be removed.

Reply:
removed (on the cast)